.github/workflows/ci.yml

```diff
@@ -0,0 +1,331 @@
+          file: '${{ env.file }}'
+          context: '${{ env.run_from }}'
+  build:
+    runs-on: 'ubuntu-latest'
+    needs: [build-lumberjack-server, build-go-grpc-client, build-go-http-client, build-java-grpc-client, build-java-http-client]
+    steps:
+      - run: 'echo build complete'
+
+  # Deploy integration test infrastructure and run integration tests
+  integration:
+    runs-on: 'ubuntu-latest'
+    needs: [build]
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    env:
+      tf_module_dir: './terraform/modules/ci'
+    steps:
+      - name: 'Checkout'
+        uses: 'actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b' # ratchet:actions/checkout@v3
+      - id: 'auth'
+        name: 'Authenticate to Google Cloud'
+        uses: 'google-github-actions/auth@ceee102ec2387dd9e844e01b530ccd4ec87ce955' # ratchet:goe-github-actions/auth@v0
+        with:
+          workload_identity_provider: '${{ env.wif_provider }}'
+          service_account: '${{ env.wif_service_account }}'
+          token_format: 'access_token'
+
+      # TODO: Workload Identity doesn't work with GCS Provider.
+      # https://github.com/google-github-actions/auth/issues/89
+      - name: 'Workaround Terraform issue'
+        shell: 'bash'
+        run: |-
+          echo "GOOGLE_OAUTH_ACCESS_TOKEN=${{ steps.auth.outputs.access_token }}" >> $GITHUB_ENV
+      - name: 'Setup Terraform'
+        uses: hashicorp/setup-terraform@17d4c9b8043b238f6f35641cdd8433da1e6f3867 # ratchet:hashicorp/setup-terraform@v2
+        with:
+          terraform_wrapper: false
+      - name: 'Write tfvars file'
+        env:
+          client_images: '\{\"ljc-go-grpc\":\"${{env.docker_repo}}/ljc-go-grpc:${{env.docker_tag}}\",\"ljc-go-http\":\"${{env.docker_repo}}/ljc-go-http:${{env.docker_tag}}\",\"ljc-java-grpc\":\"${{env.docker_repo}}/ljc-java-grpc:${{env.docker_tag}}\",\"ljc-java-http\":\"${{env.docker_repo}}/ljc-java-http:${{env.docker_tag}}\"\}'
+          server_image: '${{env.docker_repo}}/lumberjack-server:${{env.docker_tag}}'
+        run: |-
+            var_file=/tmp/ci.tfvars
+
+            touch /tmp/ci.tfvars
+            echo commit_sha=\"${{ env.docker_tag }}\" >> ${var_file};
+            echo server_project_id=\"${{ env.server_project_id }}\" >> ${var_file};
+            echo server_image=\"${{ env.server_image }}\" >> ${var_file};
+            echo server_service_name=\"lumberjack-server\" >> ${var_file};
+            echo client_project_id=\"${{ env.client_project_id }}\" >> ${var_file};
+            echo client_images=${{ env.client_images }} >> ${var_file};
+            echo >> ${var_file};
+
+            cat ${var_file};
+            echo "tfvar_file=${var_file}" >> $GITHUB_ENV;
+      - name: 'Init the terraform infrastructure'
+        run: terraform -chdir=${{ env.tf_module_dir }} init
+      - name: 'Apply the infrastructure updates'
```

Comment:
IIUC, we are not using any Cloud Run revision tricks now, right? Meaning we will create new Cloud Run service in each run.

Reply:
That's correct. I'll follow up with that change in a future PR.

Reply:
Got it. TBH if it doesn't cause a significant difference in the CI running time, I don't care if it's a new service vs. revision.